### PR TITLE
Sprint 2/issue 19#places list screen design

### PR DESCRIPTION
**Pull request by ramondiaz21**
# Description

What did you implemented/Why did you implemented this:

- I made the view where the places will be shown, all the interactive part.
- Responsive was added too.
- It includes controller, view and the route.
- I made several examples, to see how will work once the crud were done.

## Testing

- Just run the server, and enter in the /places/list route.

## Screenshots

image

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -166,32 +171,32 @@ label {
...
-.container{
+/* .container{
   display: flex;
   flex-direction: column;
   justify-content: center;
   align-items: center;
   width: 100%;
   padding: 50px 0;
-}
+} */
```
Comment (JuanequeX) [inline app/assets/stylesheets/application.css:193]: Remove this commented code

File: app/assets/stylesheets/places.css [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+  .slider-items {
+    display: flex;
+    gap: 25px;
+    transition: transform 0.5s ease;
+  }
+
+  .slider-item {
+    align-items: center;
+    align-items: center;
```
Comment (JuanequeX) [inline app/assets/stylesheets/places.css:26]:
```suggestion
    align-items: center;
```

File: app/assets/stylesheets/places.css [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    display: flex;
+    flex-direction: column;
+    gap: 10px;
+    justify-content: center;
+    justify-content: center;
```
Comment (JuanequeX) [inline app/assets/stylesheets/places.css:31]:
```suggestion
    justify-content: center;
```

File: app/assets/stylesheets/places.css [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    border: 1px solid black;
+    border-radius: 100px;
+    color: black;
+    cursor: pointer;
+    display: flex;
+    font-size: 33px;
+    height: 29px;
+    justify-content: center;
+    padding: 10px;
+    padding: 8px;
```
Comment (JuanequeX) [inline app/assets/stylesheets/places.css:51]: in this case the `padding` applied is `8px` if is correct, remove the above line.

File: app/controllers/places_controller.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class PlacesController < ApplicationController
+  def list
+  end
```
Comment (JuanequeX) [inline app/controllers/places_controller.rb:3]:
```suggestion
  def list; end
```

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -7,7 +8,12 @@
     <%= csp_meta_tag %>
     <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbo-track': 'reload' %>
-    <%= javascript_importmap_tags %>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet"
```
Comment (JuanequeX) [inline app/views/layouts/application.html.erb:11]: Improvements can be made to the integration of Bootstrap by leveraging the impormaps of Rails to incorporate external libraries more effectively. Actually, my suggestion is to use the Rails approach.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -24,23 +30,14 @@
...
-            crossorigin="anonymous"></script>
+            integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
     <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.min.js"
-            integrity="sha384-cVKIPhGWiC2Al4u+LWgxfKTRIcfu0JTxR+EQDz/bgldoEyl4H0zUF0QKbrJ0EcQF"
-            crossorigin="anonymous"></script>
+            integrity="sha384-cVKIPhGWiC2Al4u+LWgxfKTRIcfu0JTxR+EQDz/bgldoEyl4H0zUF0QKbrJ0EcQF" crossorigin="anonymous"></script>
     <script src="https://kit.fontawesome.com/a991b50074.js" crossorigin="anonymous"></script>
+    <%= javascript_importmap_tags %>
+    <%= javascript_include_tag 'sliders.js', 'data-turbolinks-track': 'reload' %>
```
Comment (JuanequeX) [inline app/views/layouts/application.html.erb:41]: Look, this js tag can be deleted and integrated  this one or any other library how `swiper` through the impormaps

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/assets/javascript/sliders.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    prevButton.style.display = "flex";
+  }
+
+  // Mostrar u ocultar el botón de "Next" dependiendo de la posición actual
+  if (currentIndex >= items.length - 10) {
+    nextButton.style.display = "none";
+  } else {
+    nextButton.style.display = "flex";
+  }
+}
```
Comment (juperameza) [inline app/assets/javascript/sliders.js:38]: In my humble opinion I think comments like this just bloat your code, I thinks it intuitive what the code does if not you should aim to improve your code rather than commenting

File: app/assets/stylesheets/places.css [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+  .like-this svg {
+    height: 25px; /* Alto del SVG */
+    width: 25px; /* Ancho del SVG */
+  }
+
+  .like-this path {
+    fill: rgba(0, 0, 0, 0.5); /* Relleno negro con 50% de transparencia */
+    stroke: #fff; /* Borde blanco */
+    stroke-width: 1px; /* Ancho del borde */
+  }
```
Comment (juperameza) [inline app/assets/stylesheets/places.css:258]: Se why are you commenting this ??

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/assets/config/manifest.js [hunk 10]
```diff
@@ -3,3 +3,4 @@
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
 //= link_tree ../builds
+//= link sliders.js
```
Comment (MauricioNC) [inline app/assets/config/manifest.js:6]: Add a new line at the end of your code

**Review by MauricioNC — CHANGES_REQUESTED** [review]

**Review by MauricioNC — APPROVED** [review]

**Review by juperameza — APPROVED** [review]

**Review by AnaCarrizales — APPROVED** [review]

**Review by juperameza — APPROVED** [review]

**Review by JuanequeX — APPROVED** [review]
